lib/src/platform_dropdown_button.dart

```diff
@@ -0,0 +1,217 @@
+    final data = material?.call(context, platform(context));
+    return DropdownButton(
+      //Common
+      key: data?.widgetKey ?? widgetKey,
+      items: data?.items ?? items,
+      onChanged: data?.onChanged ?? onChanged,
+      value: data?.value ?? value,
+
+      //Material only
+      selectedItemBuilder: data?.selectedItemBuilder,
+      hint: data?.hint,
+      disabledHint: data?.disabledHint,
+      onTap: data?.onTap,
+      elevation: data?.elevation ?? 8,
+      style: data?.style,
+      underline: data?.underline,
+      icon: data?.icon,
+      iconDisabledColor: data?.iconDisabledColor,
+      iconEnabledColor: data?.iconEnabledColor,
+      iconSize: data?.iconSize ?? 24.0,
+      isDense: data?.isDense ?? false,
+      isExpanded: data?.isExpanded ?? false,
+      itemHeight: data?.itemHeight ?? kMinInteractiveDimension,
+      focusColor: data?.focusColor,
+      focusNode: data?.focusNode,
+      autofocus: data?.autofocus ?? false,
+      dropdownColor: data?.dropdownColor,
+      menuMaxHeight: data?.menuMaxHeight,
+      enableFeedback: data?.enableFeedback,
+      alignment: data?.alignment ?? AlignmentDirectional.centerStart,
+      borderRadius: data?.borderRadius,
+      padding: data?.padding,
+    );
+  }
+
+  @override
+  CupertinoButton createCupertinoWidget(BuildContext context) {
+    final data = cupertino?.call(context, platform(context));
+    void _showCupertinoPicker() {
+      showCupertinoModalPopup<void>(
+        context: context,
+        builder: (BuildContext context) => Container(
+          height: 216,
+          padding: const EdgeInsets.only(top: 6.0),
+          // The Bottom margin is provided to align the popup above the system navigation bar.
+          margin: EdgeInsets.only(
+            bottom: MediaQuery.of(context).viewInsets.bottom,
+          ),
+          // Provide a background color for the popup.
+          color: CupertinoColors.systemBackground.resolveFrom(context),
+          // Use a SafeArea widget to avoid system overlaps.
+          child: SafeArea(
+            top: false,
+            child: CupertinoPicker(
+              itemExtent: 32.0,
+              onSelectedItemChanged: (int index) {
+                if (data?.onChanged != null)
+                  data?.onChanged(items[index].value);
+                else
+                  onChanged(items[index].value);
+              },
+              children: data?.items != null
+                  ? data!.items.map((e) => e.child).toList()
+                  : items.map((e) => e.child).toList(),
+            ),
+          ),
+        ),
+      );
+    }
+
+    return CupertinoButton(
+      // Common
+      key: data?.widgetKey ?? widgetKey,
+      onPressed: () {
+        _showCupertinoPicker();
+      },
+      child: data?.value != null
+          ? Text('${data!.value}')
+          : value != null
+              ? Text('$value')
+              : Text('Cupertino'),
+    );
+  }
```

Comment:
Issue with this is that it is very opinionated to having used `CupertinoButton`.